src/av/errors.rs

```diff
@@ -0,0 +1,286 @@
+use std::error::Error;
+use std::fmt;
+
+///////////////////////////////
+// Creation and destruction //
+/////////////////////////////
+/// Creation and destruction
+#[repr(C)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum NewError {
```

Comment:
Change `TOXAV_ERR_NEW → NewError` can cause confusion - IMHO it would be beneficial to prepend it with `Av`, so that it would be clear for anyone who's even briefly looking at the code that it's about A/V.